x/merkledb/trie.go

```diff
@@ -92,11 +92,10 @@ type View interface {
 	CommitToDB(ctx context.Context) error
 }
 
-// Returns the nodes along the path to [key].
+// Calls [visitNode] on the nodes along the path to [key].
 // The first node is the root, and the last node is either the node with the
 // given [key], if it's in the trie, or the node with the largest prefix of
 // the [key] if it isn't in the trie.
-// Always returns at least the root node.
```

Comment:
why was this dropped?

Reply:
It used to be true, but it isn't true now. If the root is not a prefix of the key we're looking for, the root will not be visited.